Let the compiler generate more operator==() in WTF/

cdumez opened the pull request:
#### fc1741dbad51fcbfcc7b2a85387c97e29caaa03b
<pre>
Let the compiler generate more operator==() in WTF/
<a href="https://bugs.webkit.org/show_bug.cgi?id=260840">https://bugs.webkit.org/show_bug.cgi?id=260840</a>

Reviewed by Alan Baradlay and Darin Adler.

Let the compiler generate more operator==() in WTF/ now that we allow C++20.

* Source/WTF/wtf/Bag.h:
* Source/WTF/wtf/CodePtr.h:
(WTF::CodePtr::operator== const): Deleted.
* Source/WTF/wtf/CompactPointerTuple.h:
* Source/WTF/wtf/CountingLock.h:
* Source/WTF/wtf/DateMath.h:
(WTF::LocalTimeOffset::operator== const): Deleted.
* Source/WTF/wtf/FixedBitVector.h:
(WTF::FixedBitVector::operator== const): Deleted.
* Source/WTF/wtf/FunctionPtr.h:
* Source/WTF/wtf/GenericTimeMixin.h:
(WTF::GenericTimeMixin::operator== const): Deleted.
* Source/WTF/wtf/LikelyDenseUnsignedIntegerSet.h:
(WTF::LikelyDenseUnsignedIntegerSet::iterator::operator== const): Deleted.
* Source/WTF/wtf/ListHashSet.h:
(WTF::ListHashSetIterator::operator== const): Deleted.
* Source/WTF/wtf/OptionSet.h:
(WTF::OptionSet::Iterator::operator== const): Deleted.
(WTF::OptionSet::operator==): Deleted.
* Source/WTF/wtf/OrderMaker.h:
(WTF::OrderMaker::iterator::operator== const): Deleted.
* Source/WTF/wtf/Range.h:
(WTF::Range::operator== const): Deleted.
* Source/WTF/wtf/RefVector.h:
(WTF::RefVectorIterator::operator== const): Deleted.
(WTF::RefVectorConstIterator::operator== const): Deleted.
* Source/WTF/wtf/ReferenceWrapperVector.h:
(WTF::ReferenceWrapperVectorIterator::operator== const): Deleted.
(WTF::ReferenceWrapperVectorConstIterator::operator== const): Deleted.
* Source/WTF/wtf/Seconds.h:
* Source/WTF/wtf/SentinelLinkedList.h:
(WTF::SentinelLinkedList::BaseIterator::operator== const): Deleted.
* Source/WTF/wtf/SingleRootGraph.h:
(WTF::SingleRootGraphNode::operator== const): Deleted.
* Source/WTF/wtf/TimeWithDynamicClockType.h:
* Source/WTF/wtf/UUID.h:
(WTF::UUID::operator== const): Deleted.
* Source/WTF/wtf/UniqueRefVector.h:
(WTF::UniqueRefVectorIterator::operator== const): Deleted.
(WTF::UniqueRefVectorConstIterator::operator== const): Deleted.
* Source/WTF/wtf/text/CodePointIterator.h:
(WTF::CodePointIterator::operator== const): Deleted.
* Source/WTF/wtf/text/OrdinalNumber.h:
(WTF::OrdinalNumber::operator== const): Deleted.
* Source/WTF/wtf/text/TextPosition.h:
(WTF::TextPosition::operator== const): Deleted.

Canonical link: <a href="https://commits.webkit.org/267427@main">https://commits.webkit.org/267427@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/36e204fc2826e1ca42c60712a0d40374b8c48195

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/16516 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/16838 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/17278 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/18297 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/15490 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/20069 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/16981 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/17836 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/16713 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/17123 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/14284 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/19075 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/14367 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/14968 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/21763 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/14251 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/15355 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/15134 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/19447 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/15752 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/15730 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/13355 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/18073 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/14930 "Built successfully") | | [  ~~🧪 jsc-armv7-tests~~](https://ews-build.webkit.org/#/builders/25/builds/4251 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/3966 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/19299 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/19293 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/15555 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/4083 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WTF/wtf/OptionSet.h:
```diff
@@ -205,10 +205,7 @@ template<typename E> class OptionSet {
-    {
-        return lhs.m_storage == rhs.m_storage;
-    }
+    constexpr bool operator==(const OptionSet& other) const = default;
```
No need for an argument name here.

darinadler commented on Source/WTF/wtf/ReferenceWrapperVector.h:
```diff
@@ -48,7 +48,7 @@ class ReferenceWrapperVectorIterator {
     T* operator->() const { return m_iterator->ptr(); }
 
-    bool operator==(const Iterator& other) const { return m_iterator == other.m_iterator; }
+    bool operator==(const Iterator& other) const = default;
```
No need for an argument name here.

darinadler commented on Source/WTF/wtf/ReferenceWrapperVector.h:
```diff
@@ -84,7 +84,7 @@ class ReferenceWrapperVectorConstIterator {
     const T* operator->() const { return m_iterator->ptr(); }
 
-    bool operator==(const Iterator& other) const { return m_iterator == other.m_iterator; }
+    bool operator==(const Iterator& other) const = default;
```
No need for an argument name here.

darinadler commented on Source/WTF/wtf/SentinelLinkedList.h:
```diff
@@ -107,10 +107,7 @@ template <typename T, typename RawNode = T> class SentinelLinkedList {
-        {
-            return m_node == other.m_node;
-        }
+        bool operator==(const BaseIterator& other) const = default;
```
No need for an argument name here.

darinadler commented on Source/WTF/wtf/text/OrdinalNumber.h:
```diff
@@ -41,7 +41,7 @@ class OrdinalNumber {
     int oneBasedInt() const { return m_zeroBasedValue + 1; }
 
-    bool operator==(OrdinalNumber other) const { return m_zeroBasedValue == other.m_zeroBasedValue; }
+    bool operator==(const OrdinalNumber& other) const = default;
```
No need for an argument name here.

darinadler reviewed (approved):
Did you consider the `friend` form, instead of the member function? I ask because that allows conversions in the left side of `==`, which might occasionally be helpful. It’s a little wordier because you have to utter the class name twice.

cdumez committed:
Let the compiler generate more operator==() in WTF/
https://bugs.webkit.org/show_bug.cgi?id=260840

Reviewed by Alan Baradlay and Darin Adler.

Let the compiler generate more operator==() in WTF/ now that we allow C++20.

* Source/WTF/wtf/Bag.h:
* Source/WTF/wtf/CodePtr.h:
(WTF::CodePtr::operator== const): Deleted.
* Source/WTF/wtf/CompactPointerTuple.h:
* Source/WTF/wtf/CountingLock.h:
* Source/WTF/wtf/DateMath.h:
(WTF::LocalTimeOffset::operator== const): Deleted.
* Source/WTF/wtf/FixedBitVector.h:
(WTF::FixedBitVector::operator== const): Deleted.
* Source/WTF/wtf/FunctionPtr.h:
* Source/WTF/wtf/GenericTimeMixin.h:
(WTF::GenericTimeMixin::operator== const): Deleted.
* Source/WTF/wtf/LikelyDenseUnsignedIntegerSet.h:
(WTF::LikelyDenseUnsignedIntegerSet::iterator::operator== const): Deleted.
* Source/WTF/wtf/ListHashSet.h:
(WTF::ListHashSetIterator::operator== const): Deleted.
* Source/WTF/wtf/OptionSet.h:
(WTF::OptionSet::Iterator::operator== const): Deleted.
(WTF::OptionSet::operator==): Deleted.
* Source/WTF/wtf/OrderMaker.h:
(WTF::OrderMaker::iterator::operator== const): Deleted.
* Source/WTF/wtf/Range.h:
(WTF::Range::operator== const): Deleted.
* Source/WTF/wtf/RefVector.h:
(WTF::RefVectorIterator::operator== const): Deleted.
(WTF::RefVectorConstIterator::operator== const): Deleted.
* Source/WTF/wtf/ReferenceWrapperVector.h:
(WTF::ReferenceWrapperVectorIterator::operator== const): Deleted.
(WTF::ReferenceWrapperVectorConstIterator::operator== const): Deleted.
* Source/WTF/wtf/Seconds.h:
* Source/WTF/wtf/SentinelLinkedList.h:
(WTF::SentinelLinkedList::BaseIterator::operator== const): Deleted.
* Source/WTF/wtf/SingleRootGraph.h:
(WTF::SingleRootGraphNode::operator== const): Deleted.
* Source/WTF/wtf/TimeWithDynamicClockType.h:
* Source/WTF/wtf/UUID.h:
(WTF::UUID::operator== const): Deleted.
* Source/WTF/wtf/UniqueRefVector.h:
(WTF::UniqueRefVectorIterator::operator== const): Deleted.
(WTF::UniqueRefVectorConstIterator::operator== const): Deleted.
* Source/WTF/wtf/text/CodePointIterator.h:
(WTF::CodePointIterator::operator== const): Deleted.
* Source/WTF/wtf/text/OrdinalNumber.h:
(WTF::OrdinalNumber::operator== const): Deleted.
* Source/WTF/wtf/text/TextPosition.h:
(WTF::TextPosition::operator== const): Deleted.

Canonical link: https://commits.webkit.org/267427@main
